packages/realm-react/src/runtime.d.ts

```diff
@@ -0,0 +1,27 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+declare type Console = {
+  log(...args: unknown[]): void;
+  error(...args: unknown[]): void;
+  warn(...args: unknown[]): void;
+};
+
+declare const console: Console;
+
+declare function setImmediate(cb: (...args: unknown[]) => void);
```

Comment:
I suggest adding this to declare the minimal runtime APIs that we allow the package to access, ensuring we implement against APIs that are supported by both Node.js and React Native.

Reply:
Does this get picked up automatically?  Does this ensure methods like `setTimeout` need to be typed here before they are used?

Reply:
Exactly. It gets picked up because the root tsconfig.json has `src` in includes and since the file isn't using ESM import / export statements, it's declaring ambient types and the `declare` statements here, become decalrations of globals.

Reply:
Ahh...so it could be renamed and still work.  Interesting trick.